### PR TITLE
Align Dropdown and DropdownMenu styles 

### LINKED ISSUE #8559: Dropdown: Popover content is wrong positioned
**Describe the bug**
When rendering a `Dropdown` component outside Gutenberg, the popover content is wrong positioned. 

**To Reproduce**
Execute the code below in a new React project after installing `@wordpress/components`.

```jsx
import React from 'react';
import ReactDOM from 'react-dom';
import { Button, Dropdown } from '@wordpress/components';

import '@wordpress/components/build-style/styles.css';

const MyDropdown = () => (
	<Dropdown
		position="bottom right"
		renderToggle={ ( { isOpen, onToggle } ) => (
			<Button isPrimary onClick={ onToggle } aria-expanded={ isOpen }>
				Toggle Popover!
			</Button>
		) }
		renderContent={ () => (
			<div>
				This is the content of the popover.
			</div>
		) }
	/>
);

ReactDOM.render(
	<MyDropdown />,
	document.getElementById( 'root' )
);
```

**Expected behavior**
The popover content should be displayed below the button.

**Screenshots**
screen shot 2018-08-05 at 22 49 24

**Desktop:**
 - OS: macOS High Sierra
 - Browser Chrome
 - Version 67

**Additional context**
Issue found while working on #8338 and Automattic/wp-calypso#26367

### PR DESCRIPTION
**Pull request by youknowriad**
Closes #8559
While working on #18624 I notice that if I use Dropdown or DropdownMenu, the styling of the DropdownToggle is different. This PR tries to address that by:

 - Removing specific styles from DropdownMenu and just relying on the default styles of IconButton
 - In addition to that I automatically applied inline-block to all Dropdown wrappers, i think it makes sense IMO (duplicate of #16491). this allows the popover to be positioned properly.

The downside here is that the specific styles for the Toggle in DropdownMenu were specifically targetted to match the styling of `ToolbarButton` (block toolbars) which means the size of the regular buttons and the dropdowns (hover) is not consistent now. My thinking at the moment is that we should fix this by removing the specific styles for `ToobarButton` and instead make sure any `IconButton` (ToolbarButton or not) inside `Toolbar` component should be adapted (hover styles with a smaller size). 

Capture d’écran 2019-11-20 à 2 25 13 PM

I decided to leave the toolbar hover style fix for a separate PR but if you think the breakage in the current PR is severe enough, I'm happy to do everything in the same PR. I preferred not including here to keep the size of the PR smallish.

I also added a temporary story to the Dropdown component where we show both a DropdownMenu and a Dropdown in the same canvas to be able to compare them.

Capture d’écran 2019-11-20 à 2 25 27 PM

### CODE REVIEW

File: packages/components/src/dropdown/style.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.components-dropdown {
+	display: inline-block;
```
Comment (ItsJonQ) [inline packages/components/src/dropdown/style.scss]: Thanks for fixing this! Changing the display of the `div` makes sense. Can confirm it's working in Storybook 👍

**Review by ItsJonQ — APPROVED** [review]
@youknowriad Tested in Docker + Storybook, and it's working as expected. Awesome job 🚀 .

> I decided to leave the toolbar hover style fix for a separate...

I agree 👍

**Comment by youknowriad** [comment]
When trying to work on the toolbar buttons style alignments, I noticed that it basically comes down to remove styles from the `ToolbarButton` component, so I just went ahead and did it here.

The hover style is not consistent but bigger than it was before. It could be considered a very small visual regression but I think consistency wins here and if we were to restore the smaller hover style, we should restore it for all components consistently.

**Comment by jasmussen** [comment]
This is what I see when running this branch:

dropdowns

If that's also what you see, then 👍 👍

**Comment by gziolo** [comment]
I noticed that this change created regression for toolbar buttons:
toggled-buttons

When they are toggled they no longer change their visual appearance. This is how it works before 7.0:

Screen Shot 2019-11-29 at 15 38 14

**Pull request merged**